### DISCUSSION TITLE
Saving multiple graphs into one file

### DISCUSSION THREAD

**Discussion by adam2392** [networkx/networkx#5611] [Q&A] [answered]
Hi,

If I'm working on a downstream package that utilizes networkx graphs under the hood, is there a recommended way I can store my object but keep all the graphs in one file? Is there such a mechanism? 

Thanks!

**Comment by dschult** [comment, marked as answer]
It probably depends on what you are going to do with these files, but pickle is  a standard Python way to store objects. 
Just make a list of your graphs, dump it to a pickle file and when you need to, load it back in (unpickle it).

**Reply by rossbar** [reply]
> Is there such a mechanism?

Just to second @dschult 's answer - there are many such mechanisms! I personally probably wouldn't turn to `pickle` first given the portability concerns (though I have no idea if that's still an issue these days). I'd probably start with json, but only because having text files is nice. If size on disk is a concern that's probably not the right way to go... so as mentioned, it really depends on your own needs/goals!
